### DISCUSSION TITLE
how to solve this err: CMake Error at cmake/python.cmake:197 (message): Python executable has not been found

### DISCUSSION THREAD

**Discussion by tjuzhcs** [stack-of-tasks/pinocchio#2533] [Q&A] [answered]
I want to install pinocchio, but meet this problem:

CMake Error at cmake/python.cmake:197 (message):
  Python executable has not been found.
Call Stack (most recent call first):
  CMakeLists.txt:303 (findpython)

Does anyone know how to do?

**Comment by ManifoldFR** [comment]
Hi, thanks for your interest in Pinocchio! I'll ask a few questions for further info:

What is your operating system?

Against what kind of environment are you trying to build Pinocchio ? A conda env, a virtualenv, system (i.e. installed through apt, pacman...) Python ?

What CMake command did you run ?

**Comment by jcarpent** [comment, marked as answer]
Normally, by setting the CMake variable `PYTHON_EXECUTABLE` when running the cmake command should solve your issue:
```bash
cmake . -DPYTHON_EXECUTABLE=$(which python3)
```
